classes/task/sendfailednotifications_task.php

```diff
@@ -61,58 +70,55 @@ private function get_failed_rows() {
     private function get_failed_email_message($results) {
         $emailmsg = "";
 
-        $emailmsg .= "<p>".PHP_EOL;
-        $emailmsg .= get_string('failedtosend', 'logstore_xapi').PHP_EOL;
-        $emailmsg .= "</p>".PHP_EOL;
-
+        // styles
+        $emailmsg .= '<style type="text/css">.header {text-align:left;}</style>';
+    
+        // first line
+        $emailmsg .= html_writer::tag('p', get_string('failedtosend', 'logstore_xapi'));
+    
         // summary info
-        $emailmsg .= "<table>".PHP_EOL;
-
         $endpointname = get_string('endpoint', 'logstore_xapi');
         $url = get_config('logstore_xapi', 'endpoint');
-        $endpointurl = "<a href='$url' target='_blank'>$url</a>";
-
+        $endpointurl = html_writer::tag('a', $url, array('target' => '_blank', 'href' => $url));
+    
         $errorlogpage = get_string('errorlogpage', 'logstore_xapi');
         $url = new \moodle_url("/admin/tool/log/store/xapi/report.php");
-        $errorlogurl = "<a href='$url' target='_blank'>$url</a>";
-
-        $row = "<tr>";
-        $row .= "<td>".$endpointname."</td>";
-        $row .= "<td>".$endpointurl."</td>";
-        $row .= "</tr>".PHP_EOL;
-        $emailmsg .= $row;
-
-        $row = "<tr>";
-        $row .= "<td>".$errorlogpage."</td>";
-        $row .= "<td>".$errorlogurl."</td>";
-        $row .= "</tr>".PHP_EOL;
-        $emailmsg .= $row;
-
-        $emailmsg .= "</table>".PHP_EOL;
-
+        $errorlogurl = html_writer::tag('a', $url, array('target' => '_blank', 'href' => $url));
+    
+        // first table
+        $table = new html_table();
+    
+        // data
+        $table->data[] = array($endpointname, $endpointurl);
+        $table->data[] = array($errorlogpage, $errorlogurl);
+    
+        // add table to message
+        $emailmsg .= html_writer::table($table);
+    
         // separator
-        $emailmsg .= "<h3>".get_string('failurelog', 'logstore_xapi')."</h3>".PHP_EOL;
-
-        // rows
-        $emailmsg .= "<table>".PHP_EOL;
-
-        $emailmsg .= '<tr>'.PHP_EOL;
-        $emailmsg .= "<th align='left'>".get_string('datetimegmt', 'logstore_xapi')."</th>".PHP_EOL;
-        $emailmsg .= "<th align='left'>".get_string('eventname', 'logstore_xapi')."</th>".PHP_EOL;
-        $emailmsg .= "<th align='left'>".get_string('response', 'logstore_xapi')."</th>".PHP_EOL;
-        $emailmsg .= "</tr>".PHP_EOL;
-
+        $emailmsg .= html_writer::tag('h3', get_string('failurelog', 'logstore_xapi'));
+    
+        // second table
+        $table = new html_table();
+        $table->attributes['style'] = "align:left";
```

Comment:
instead of doing this you can do this:
$table->align = array (null, 'left', 'right');